overlord/snapstate/snapmgr_test.go

```diff
@@ -3105,6 +3117,158 @@ func (s *canRemoveSuite) TestActiveOSAndKernelAreNotOK(c *C) {
 	c.Check(snapstate.CanRemove(kernel, true), Equals, false)
 }
 
+func revs(seq []*snap.SideInfo) []int {
+	revs := make([]int, len(seq))
+	for i, si := range seq {
+		revs[i] = si.Revision.N
+	}
+
+	return revs
+}
+
+// build a SnapState with a revision sequence given by `before` and a
+// current revision of `current`. Then refresh --revision via. Then
+// check the revision sequence is as in `after`.
+func (s *snapmgrTestSuite) testOpSequence(c *C, revert bool, before []int, current, via int, after []int) (*snapstate.SnapState, *state.TaskSet) {
```

Comment:
can this thing take a struct, I'm already lost trying to read the tests belows, is just numbers